### ISSUE TITLE
SvelteKit 2: New requirement: "Dynamic environment variables cannot be used during prerendering" causes build time errors

### ISSUE THREAD

**Issue by ar-rm** [baisong/sveltekit-airtable-quickstart#1]
Hi,

This code is great! It really helped me with a SvelteKit 1.x project that uses Airtable as the primary database.

Thank you!

---

#### tl;dr: SvelteKit 2's new requirement that "Dynamic environment variables cannot be used during prerendering" causes build time errors with this code.

If you (or anyone using this code) upgrades to SvelteKit 2.x [^1], it looks like it may be necessary to switch from `dynamic` to *explicitly* declared `static` environment variables in `/src/routes/airtable/+page.server.js`

[^1]: `"@sveltejs/kit": "next",` is set in the [package.json](https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/package.json#L15) file. So it seems like anyone using this code, from December 14, 2023 on, will be affected.

---

I upgraded to [SvelteKit 2.0](https://svelte.dev/blog/sveltekit-2). Running `npm run dev` continues to work as expected. But `npm run build` failed with the following error:

❌ `Error: Cannot read values from $env/dynamic/private while prerendering (attempted to read env.AIRTABLE_KEY). Use $env/static/private instead`

After some research, it looks like it's related to this specific SK 2 change:

[Dynamic environment variables cannot be used during prerendering](https://kit.svelte.dev/docs/migrating-to-sveltekit-2#dynamic-environment-variables-cannot-be-used-during-prerendering)

> Because of this, dynamic environment variables can no longer be read during prerendering in SvelteKit 2 — you should use the static modules instead.

(I have `export const prerender = true;` set in my `/+layout.js` file. And I see that in your [/src/routes/+page.js](https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/src/routes/%2Bpage.js#L3) file.)

This is the line in your code that I changed in my file:

https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/src/routes/airtable/%2Bpage.server.js#L2

Before:
`import { env } from '$env/dynamic/private';`

After:
`import { env } from '$env/static/private';`

However that doesn't immediately work. I then receive this error:

❌ `error during build: RollupError: "env" is not exported by "virtual:$env/static/private", imported by "src/routes/airtable/+page.server.js".`

After some more research, I found the following post:
https://github.com/sveltejs/kit/issues/11378#issuecomment-1861074759

Which reminded me that ***static* private** `env` variables need to be declared explicitly. So unlike `dynamic` imports, you can't import `env` and then chain onto it. Ex: `env.AIRTABLE_KEY` = ❌

From: https://kit.svelte.dev/docs/modules#$env-static-private
> Unlike [$env/dynamic/private](https://kit.svelte.dev/docs/modules#$env-dynamic-private), the values exported from this module are statically injected into your bundle at build time, enabling optimisations like dead code elimination.
>
> [...]
>
> Note that all environment variables referenced in your code should be declared (for example in an .env file), even if they don't have a value until the app is deployed

Which lead me to updating the following places (in my code):

https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/src/routes/airtable/%2Bpage.server.js#L2

https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/src/routes/airtable/%2Bpage.server.js#L4

https://github.com/baisong/sveltekit-airtable-quickstart/blob/00decf45a1f3325828cfac190aa8bb6a09f3cfb0/src/routes/airtable/%2Bpage.server.js#L25

Before:
```
import { env } from '$env/dynamic/private';

const myBaseId = import.meta.env.VITE_AIRTABLE_BASE_ID;
apiKey: env.AIRTABLE_KEY,
```

After:
```
import { VITE_AIRTABLE_BASE_ID } from '$env/static/private';
import { AIRTABLE_KEY } from '$env/static/private';

const myBaseId = VITE_AIRTABLE_BASE_ID;
apiKey: AIRTABLE_KEY,
```

✅ No more errors during `npm run build`

(It also looks like you no longer need to use the Vite specific `import.meta.env` prefix.)

---

I was tempted to submit a Pull Request, but didn't because:
1. I'm not an expert and I may be wrong about some of this.
2. I'm not totally sure this SvelteKit 2.x change *needs* to be made to *your* example code. Maybe it only needed to be made because of project-specific changes I've made.

So I thought it would be better to open this Issue first.

But if a Pull Request *is* the correct thing to do, just let me know :-)

---

I hope this helps.

Thanks again!

**Comment by baisong** [comment]
Thank you so much @ar-rm for documenting this issue and a fix! I just ran into this same issue and your solution worked for me.

I'd love if you wanted to submit a PR, otherwise I will get around to it sometime!
